src/VirtoCommerce.Platform.Web/Security/Authentication/BasicAuthenticationHandler.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using VirtoCommerce.Platform.Core.Security;
+
+namespace VirtoCommerce.Platform.Web.Security.Authentication
+{
+    public class BasicAuthenticationHandler : AuthenticationHandler<BasicAuthenticationOptions>
+    {
+
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly SignInManager<ApplicationUser> _signInManager;
+
+        public BasicAuthenticationHandler(
+            IOptionsMonitor<BasicAuthenticationOptions> options,
+            ILoggerFactory logger,
+            UrlEncoder encoder,
+            ISystemClock clock,
+            UserManager<ApplicationUser> userManager,
+            SignInManager<ApplicationUser> signInManager) : base(options, logger, encoder, clock)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+        }
+
+        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
+        {
+            var authHeader = Request.Headers["Authorization"].ToString();
+            if (authHeader != null && authHeader.StartsWith("basic", StringComparison.OrdinalIgnoreCase))
+            {
+                try
+                {
+                    var encodedAuth = authHeader.Substring("Basic ".Length).Trim();
+
+                    if (string.IsNullOrEmpty(encodedAuth))
+                    {
+                        return AuthenticateResult.Fail("Invalid authentication provided, access denied.");
+                    }
+
+                    var loginPass = DecodeUserIdAndPassword(encodedAuth);
+                    var user = await _userManager.FindByNameAsync(loginPass.Login);
+                    if (user == null)
+                    {
+                        return AuthenticateResult.Fail("Invalid email provided, access denied.");
+                    }
+
+                    if (!await _signInManager.CanSignInAsync(user))
+                    {
+                        return AuthenticateResult.Fail($"User { user.UserName } is not allowed to login.");
+                    }
+
+                    if (_userManager.SupportsUserLockout && await _userManager.IsLockedOutAsync(user))
+                    {
+                        return AuthenticateResult.Fail($"User { user.UserName } is currently locked out.");
+                    }
+
+                    if (!await _userManager.CheckPasswordAsync(user, loginPass.Password))
+                    {
+                        return AuthenticateResult.Fail("Invalid password provided, access denied.");
+                    }
```

Comment:
We should not reveal such details. The message should be the same in all cases: Invalid user name or password.